test/fork/invariant/Invariants.t.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.24;
+
+import "forge-std/Test.sol";
+
+import { Utilities } from "../../helpers/Utilities.sol";
+import { DeploymentFixtures } from "../../helpers/Fixtures.sol";
+import { NodeOperator } from "../../../src/interfaces/ICSModule.sol";
+import "../../../src/lib/QueueLib.sol";
+
+contract InvariantsBase is Test, Utilities, DeploymentFixtures {
+    function setUp() public {
+        Env memory env = envVars();
+        vm.createSelectFork(env.RPC_URL);
+        initializeFromDeployment(env.DEPLOY_CONFIG);
+    }
+}
+
+using QueueLib for QueueLib.Queue;
+
+contract CSModuleInvariants is InvariantsBase {
+    function test_keys() public {
+        uint256 noCount = csm.getNodeOperatorsCount();
+        NodeOperator memory no;
+
+        uint256 totalDepositedValidators;
+        uint256 totalExitedValidators;
+        uint256 totalDepositableValidators;
+
+        for (uint256 noId = 0; noId < noCount; noId++) {
+            no = csm.getNodeOperator(noId);
+
+            assertGe(
+                no.totalAddedKeys,
+                no.totalDepositedKeys,
+                "assert added >= deposited"
+            );
+            assertGe(
+                no.totalDepositedKeys,
+                no.totalWithdrawnKeys,
+                "assert deposited >= withdrawn"
+            );
+            assertGe(
+                no.totalVettedKeys,
+                no.totalDepositedKeys,
+                "assert vetted >= deposited"
+            );
+
+            assertGe(
+                no.totalDepositedKeys - no.totalExitedKeys,
+                no.stuckValidatorsCount,
+                "assert deposited - exited >= stuck"
+            );
+
+            assertGe(
+                no.totalAddedKeys,
+                no.depositableValidatorsCount + no.totalWithdrawnKeys,
+                "assert added >= depositable + withdrawn"
+            );
```

Comment:
This one looks strange even though the relation is correct. What do we want to check here exactly?

Reply:
that withdrawn are considered correctly